jumpscale/packages/tfgrid_solutions/chats/extend_kube.py

```diff
@@ -0,0 +1,108 @@
+from jumpscale.loader import j
+from jumpscale.sals.chatflows.chatflows import GedisChatBot, StopChatFlow, chatflow_step
+from jumpscale.sals.reservation_chatflow import deployer
+from jumpscale.clients.explorer.models import State
+from jumpscale.clients.explorer.models import K8s
+
+
+class ExtendKubernetesCluster(GedisChatBot):
+    title = "Extend Kubernetes Cluster"
+    steps = ["choose_nodes_cound", "choose_flavor", "add_public_ip", "add_nodes", "success"]
+    KUBERNETES_SIZES = K8s.SIZES
+
+    @chatflow_step(title="Choose nodes count")
+    def choose_nodes_cound(self):
+        self.master_wid = int(self.kwargs.get("master_wid", 0))
+        self.nodes_count = self.int_ask("How many nodes you want to add?", default=1, required=True, min=1)
+
+    @chatflow_step(title="Choose nodes flavor")
+    def choose_flavor(self):
+        form = self.new_form()
+        sizes = [
+            f"vCPU: {data.get('cru')}, RAM: {data.get('mru')} GiB, Disk Space {data.get('sru')} GiB"
+            for data in self.KUBERNETES_SIZES.values()
+        ]
+        cluster_size_string = form.drop_down_choice("Choose the size of your nodes", sizes, default=sizes[0])
+        form.ask()
+        self.cluster_size = sizes.index(cluster_size_string.value) + 1
+        self.node_query = self.KUBERNETES_SIZES.get(self.cluster_size)
+        self.public_ip = False
+
+    @chatflow_step(title="Public Ip")
+    def add_public_ip(self):
```

Comment:
I think we shouldn't allow public ips to be assigned to workers. @xmonader what do you think?

Reply:
We checked that with @xmonader  before developing it

Reply:
I think it's ok to support IPs on workers and set the default to False :)